server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/meta/TreeMetadata.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube.startree.meta;
+
+import org.opensearch.index.compositeindex.datacube.startree.StarTreeFieldConfiguration;
+import org.opensearch.index.compositeindex.datacube.startree.aggregators.MetricEntry;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * An interface for metadata of the star-tree
+ *
+ * @opensearch.experimental
+ */
+public interface TreeMetadata {
```

Comment:
Is this necessary ?

readStarTreeDimensions, maxLeafDocs,readSkipStarNodeCreationInDimsCount and almost everything is already specific to StarTree which defeats the purpose.